src/widgets/provider-config-dialog.tsx

```diff
@@ -219,30 +218,32 @@ export const ProviderConfigDialog: React.FC<IProviderConfigDialogProps> = ({
             </FormControl>
           )}
 
-          {selectedProvider?.requiresApiKey && (
-            <TextField
-              fullWidth
-              inputRef={apiKeyRef}
-              label="API Key"
-              type={showApiKey ? 'text' : 'password'}
-              value={apiKey}
-              onChange={e => setApiKey(e.target.value)}
-              placeholder="Enter your API key..."
-              required={selectedProvider.requiresApiKey}
-              InputProps={{
-                endAdornment: (
-                  <InputAdornment position="end">
-                    <IconButton
-                      onClick={() => setShowApiKey(!showApiKey)}
-                      edge="end"
-                    >
-                      {showApiKey ? <VisibilityOff /> : <Visibility />}
-                    </IconButton>
-                  </InputAdornment>
-                )
-              }}
-            />
-          )}
+          <TextField
```

Comment:
Wonder if we should keep `requiredApiKey` and `optionalApiKey` or `undefined`, to avoid confusion with provider that do not use the API key (e.g. `Ollama`, `ChromeAI` or `webLLM`).